nucleus/__init__.py

```diff
@@ -1144,11 +1148,14 @@ def _make_request(
         :param requests_command: requests.post, requests.get, requests.delete
         :return: response JSON
         """
-        response = self._make_request_raw(payload, route, requests_command)
+        endpoint = f"{self.endpoint}/{route}"
+
+        response = self._make_request_raw(payload, endpoint, requests_command)
+
+        if not response.ok:
+            self.handle_bad_response(endpoint, requests_command, response)
 
-        if getattr(response, "status_code") not in SUCCESS_STATUS_CODES:
-            logger.warning(response)
+        return response.json()
 
-        return (
-            response.json()
-        )  # TODO: this line fails if response has code == 404
+    def handle_bad_response(self, endpoint, requests_command, response):
```

Comment:
nice, much cleaner

Reply:
Yeah no more throwing informative error messages on the ground